torch/export/export_schema.py

```diff
@@ -0,0 +1,319 @@
+        Internal = auto()
+        External = auto()
+
+    data_location: DataLocation
+    data: Union[Buffer, ExternalBuffer]
+
+
+# This is a concrete tensor backed by storage
+@dataclass
+class Tensor:
+    # storage
+    storage: Storage
+
+    # metadata
+    meta: TensorMeta
+
+
+################################################################################
+# Following section is the defining the schema of 3 level construct: GraphModule, Graph, Node
+
+# IValue has no corresponding class in fx
+# IValue is the "values" that are passed between nodes in the graph
+# !!! Assumption: Only Tensor can be passed between nodes, and not other int/float/bool types...
+# IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
+# !!! Consider using a more descriptive name, e.g. TensorValue, TensorPlaceholder, TensorArgument, etc.
+@dataclass
+class IValue:
+    name: str   # A unique identifider name for the IValue
+                # The name will be used in the graph and node to refer to the IValue
+
+    meta: TensorMeta
+
+
+@dataclass
+class NodeMetadata:
+    stack_trace: str                      # source info of a node
+    nn_module_stack: str                  # stack of nn.Module that the node originates from
+    extra: Dict[str, str]                 # arbitrary string-string pairs for extra metadata
+
+
+# Maps to fx.Node
+@dataclass
+class Node:
+    op: str         # One of ['placeholder', 'call_function', 'get_attr', 'output'],
+                    # call_method and call_module are not supported, as they shouldn't apprear in the Caononical FX Graph
+                    # !!! Consider using an enum instead of string
+
+    target: str      # fully qualified name to the target, e.g. aten.add.Tensnor
+                     # !!! Consider using a structured operator name instead of string
+
+    args: List[Argument]              # args for this node
+    kwargs: Dict[str, Argument]       # kwargs for this node
+                                      # !!! Not all types in Argument are used as kwargs, e.g. TensorArgument should not be used as kwargs
+                                      # Do we want to enforce this in the schema? i.e. only allow certain types to be used as kwargs?
+
+    outputs: List[TensorArgument]   # A list of Tensor Argument returned by this node
+                                    # !!! Notice: this assumes that a node can only return Tensor(s), and not other int/float/bool types...
+
+    metadata: NodeMetadata              # metadata fields for this node
+
+
+# Maps to fx.Graph
+@dataclass(init=False)
+class Graph:
+
+    inputs: List[Node]    # Maps to fx.graph's placeholder nodes.
+                          # Placeholder nodes are stored separately, to clearly distinguish them from other nodes.
+                          # !!! Consider making inputs a List[IValue] instead of List[Node], need to think about where to store the metadata for placeholder nodes
```

Comment:
@suo 
I very much prefer just to store graph's inputs/output as IValue, as it makes more sense. 
The question is where to store the metadata in the original input/output fx.node? Or do we care about having metadata for input/outputs node?

Reply:
Totally agree that fx's modeling on placeholder and output node is confusing. 

Notice the inconsistency on the use of node.name, node.target and node.args: 
```
opcode         name       target     args                   kwargs
-------------  ---------  ---------  ---------------------  --------
placeholder    arg0       arg0       ()                     {}
output         output     output     ([addmm_default],)     {}
```

Reply:
cc @qihqi

Reply:
What metadata would there be (besides tensor metadata, which is covered in ivalue)?

Either way, it seems low cost to add an input metadata field as needed.

Reply:
Per offline discussion, I change the graph inputs and outputs to be a List[TensorArgument].

Reply:
> What metadata would there be (besides tensor metadata, which is covered in ivalue)?
> 
> Either way, it seems low cost to add an input metadata field as needed.

TensorMeta will be stored in IValue. The remaining metadata is source information.